crates/sim2h/src/lib.rs

```diff
@@ -25,35 +24,69 @@ pub use crate::message_log::MESSAGE_LOGGER;
 use crate::{crypto::*, error::*};
 use cache::*;
 use connection_state::*;
-//use lib3h::rrdht_util::*;
 use lib3h_crypto_api::CryptoSystem;
 use lib3h_protocol::{
     data_types::{EntryData, FetchEntryData, GetListData, Opaque, SpaceData, StoreEntryAspectData},
     protocol::*,
     types::SpaceHash,
     uri::Lib3hUri,
 };
+use url2::prelude::*;
 
-use websocket::streams::{StreamEvent, StreamManager};
 pub use wire_message::{WireError, WireMessage};
 
+use in_stream::*;
 use log::*;
 use parking_lot::RwLock;
 use rand::{seq::SliceRandom, thread_rng};
 use std::{
     collections::{HashMap, HashSet},
     convert::TryFrom,
+    sync::Arc,
 };
 
+use holochain_locksmith::Mutex;
+
+pub(crate) trait MutexExt<T> {
```

Comment:
Can you document this trait? And, briefly, why is traditional mutex not good enough?